lua_examples/net_send_queue.lua

```diff
@@ -0,0 +1,46 @@
+------------------------------------------------------------------------------
+-- Net send queueing helper
+--
+-- See also
+--   https://nodemcu.readthedocs.org/en/dev/en/modules/net/#netsocketsend
+--
+-- Based on Vladimir Dronnikov's
+--   MQTT queuing publish helper
+--   https://github.com/dvv/nodemcu-thingies/blob/master/mqtt-queue-helper.lua
+--   LICENCE: http://opensource.org/licenses/MIT
+--
+-- Send arbitrary payload:
+--   local socket = net.createConnection(net.TCP, 0)
+--   socket:connect(port, ip)
+--   local net_send = dofile("net_send_queue.lua")(socket)
+--   net_send(payload1)
+--   net_send(payload2)
+--   net_send(payload3)
+--
+------------------------------------------------------------------------------
+do
+  -- cache
+  local shift = table.remove
+  -- queue handler
+  local make_sender = function(socket)
+    local queue = { }
+    local is_sending = false
+    local function send()
+      if #queue > 0 then
+        local item = shift(queue, 1)
+        socket:send(item, send)
+      else
+        is_sending = false
+      end
+    end
+    return function(...)
```

Comment:
not `...` in this case but `rec`.  Reading this would imply that we are accepting a lust.  We aren't. net:send only accepts a single string as its first argument.   Ditto on next line unless you want to add extra logic to enumerate or concat tables, etc.

Ditto next line.